### PR TITLE
add sql-function-cryptographic-functions

### LINKED ISSUE #1832: Document: feat: support sql func encrypt/decrypt
This issue tracks the documentation update needed for the merged PR #14717.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/14717
Source PR Merged At: 2024-02-01T06:56:19Z

### PR DESCRIPTION
**Pull request by ShanlanLi**
## Info

- **Description**

  - [ As title.]

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - [ https://github.com/risingwavelabs/risingwave/pull/14717]

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1832

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

**Review by WanYixian — APPROVED** [review]
LGTM

**Comment by hengm3467** [comment]
@tabVersion Please take a look. Thanks!

File: docs/sql/functions-operators/sql-function-cryptographic-functions.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+```sql title=Syntax 
+encrypt(data bytea, key bytea, type text) -> bytea
+decrypt(data bytea, key bytea, type text) -> bytea
+```  
+
+The cipher method is specified by `type`. The syntax of the `type` string is:
+
+```sql
+algorithm [ - mode ] [ /pad: padding ]
```
Comment (tabVersion) [inline docs/sql/functions-operators/sql-function-cryptographic-functions.md]: no space in square brackets
`algorithm [-mode][/pad:padding]`

**Review by tabVersion — APPROVED** [review]
rest LGTM

File: docs/sql/functions-operators/sql-function-cryptographic-functions.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+id: sql-function-cryptographic-functions
+slug: /sql-function-cryptographic-functions
+title: Cryptographic functions
+description: Cryptographic functions.
+---
+<head>
+  <link rel="canonical" href="https://docs.risingwave.com/docs/current/sql-function-cryptographic-functions/" />
+</head>
+
+### `Raw encryption functions`
```
Comment (CharlieSYH) [inline docs/sql/functions-operators/sql-function-cryptographic-functions.md:11]: I'm guessing we will offer more functions of other types in the future?

File: sidebars.js [hunk 3]
```diff
@@ -723,6 +723,11 @@ const sidebars = {
                   id: "sql/functions-operators/sql-function-comparison",
                   label: "Comparison",
                 },
+                {
+                  type: "doc",
+                  id: "sql/functions-operators/sql-function-cryptographic-functions",
+                  label: "Cryptographic functions",
```
Comment (CharlieSYH) [inline sidebars.js]:
```suggestion
                  label: "Cryptographic",
```

Comment (CharlieSYH) [reply sidebars.js]: image

File: docs/sql/functions-operators/sql-function-cryptographic-functions.md (hunk 2 above)
Comment (ShanlanLi) [reply docs/sql/functions-operators/sql-function-cryptographic-functions.md:11]: Yes, but not the near future. These two are a bit special. They are written together in PG's doc.

**Pull request merged**